### ISSUE TITLE
The parser does not support parenthesis after USING clause

### ISSUE THREAD

**Issue by radegran** [agershun/alasql#1983]
The parser does not accept left parenthesis after USING clause. The generative AI model I use for generating SQL queries keeps adding parentheses.

Example: 
```
SELECT * FROM ? a JOIN ? b USING (c)

SyntaxError: Parse error on line 1:
... ? a JOIN ? b USING (c)
-----------------------^
Expecting 'LITERAL', 'BRALITERAL', got 'LPAR'
```

[This](https://jsfiddle.net/kenbjf56/) fiddle shows the details.

**Comment by paulrutter** [comment]
I recently proposed changes to the parser, see https://github.com/AlaSQL/alasql/pull/1982. This shows it's probably not hard to support parentheses as well.

Probably here: https://github.com/AlaSQL/alasql/blob/30813b9a71c04b5e84bc8ae401b2c7cd89e68983/src/alasqlparser.jison#L2750

Add another case including LPAR and RPAR.

**Comment by mathiasrw** [comment]
Released as part of v4.6.1
